File: php/src/Services/UtilitiesService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;

class UtilitiesService
{
    private Client $client;
    public function __construct(Client $client) { $this->client = $client; }
    public function webhookResend(array $data = []): array { return $this->client->post('/webhook/resend', $data); }
    public function getTradeBlockHeight(array $data): array { return $this->client->post('/getTradeBlockHeight', $data); }
    public function checkWithdrawalAddressValidity(array $data): array { return $this->client->post('/checkWithdrawalAddressValidity', $data); }
    public function deprecatedAddress(array $data): array { return $this->client->post('/addressUnbinding', $data); }
    public function rescanLostTransaction(array $data): array { return $this->client->post('/rescanLostTransaction', $data); }
}

File: java/src/main/java/com/ccpayment/services/UtilitiesService.java
package com.ccpayment.services;

import com.ccpayment.APIError;
import com.ccpayment.Client;
import com.google.gson.JsonObject;

import java.io.IOException;

public class UtilitiesService {
    private final Client client;
    public UtilitiesService(Client client) { this.client = client; }
    public JsonObject webhookResend(JsonObject data) throws APIError, IOException { return client.post("/webhook/resend", data, JsonObject.class); }
    public JsonObject getTradeBlockHeight(JsonObject data) throws APIError, IOException { return client.post("/getTradeBlockHeight", data, JsonObject.class); }
    public JsonObject checkWithdrawalAddressValidity(JsonObject data) throws APIError, IOException { return client.post("/checkWithdrawalAddressValidity", data, JsonObject.class); }
    public JsonObject deprecatedAddress(JsonObject data) throws APIError, IOException { return client.post("/addressUnbinding", data, JsonObject.class); }
    public JsonObject rescanLostTransaction(JsonObject data) throws APIError, IOException { return client.post("/rescanLostTransaction", data, JsonObject.class); }
}
